administrator/components/com_categories/models/category.php

```diff
@@ -439,7 +440,10 @@ protected function preprocessForm(JForm $form, $data, $group = 'content')
 
 			foreach ($languages as $tag => $language)
 			{
-				if (empty($data->language) || $tag != $data->language)
+				$permission = $user->authorise('language.permission', 'com_languages.language.' . (int) $language->lang_id);
+				$canDoAssociations = !is_null($permission) ? $permission : true;
```

Comment:
To be B/C, it had to be true by default, as Rob proposed. If your last ACL patch goes in, we can test what it gives then and decide.